### PR TITLE
Broke down GUEST_START error code

### LINKED ISSUE #9909: Provide more descriptive error codes for GUEST_START
GUEST_START serves as a catch-all error for cluster node failing to start. Can minikube provide more descriptive error codes (is the failure owing to memory/cpu/disk constraints)?

### PR DESCRIPTION
**Pull request by spowelljr**
Closes #9909

Broke down GUEST_START further by adding more specific error codes
- PR_DOCKER_UNSUPPORTED
- GUEST_INCORRECT_ARCH
- GUEST_CERTS_EXPIRED
- GUEST_CNI_INCOMPATIBLE

### CODE REVIEW

File: pkg/minikube/reason/known_issues.go [hunk 1]
```diff
@@ -997,6 +1006,32 @@ var guestIssues = []match{
...
+		Regexp: re(`qemu: uncaught target signal 11 (Segmentation fault) - core dumped`),
+	},
+	{
+		Kind: Kind{
+			ID:       "GUEST_CERTS_EXPIRED",
+			ExitCode: ExGuestError,
+			Advice:   "Your minikube certs likely expired, as a workaround, clear your minikube home dir `minikube delete --all --purge`",
+			Issues:   []int{10948},
+		},
+		Regexp: re(`controlPlane never updated to`),
```
Comment (medyagh) [inline pkg/minikube/reason/known_issues.go:1025]: this one seems to Loose Regex, any better more accurate regex?

File: pkg/minikube/reason/known_issues.go [hunk 2]
```diff
@@ -267,6 +267,15 @@ var providerIssues = []match{
 		GOOS:   []string{"windows"},
 		Regexp: re(`docker:.*Mounts denied: EOF`),
 	},
+	{
+		Kind: Kind{
+			ID:       "PR_DOCKER_UNSUPPORTED",
```
Comment (medyagh) [inline pkg/minikube/reason/known_issues.go:272]: PR_DOCKER_VER_UNSUPPORTED

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/reason/known_issues.go (hunk 2 above)
Comment (spowelljr) [reply pkg/minikube/reason/known_issues.go]: Done

File: pkg/minikube/reason/known_issues.go (hunk 1 above)
Comment (spowelljr) [reply pkg/minikube/reason/known_issues.go]: Improved

File: pkg/minikube/reason/known_issues.go [hunk 3]
```diff
@@ -997,6 +1006,32 @@ var guestIssues = []match{
...
+		Regexp: re(`qemu: uncaught target signal 11 (Segmentation fault) - core dumped`),
+	},
+	{
+		Kind: Kind{
+			ID:       "GUEST_CERTS_EXPIRED",
+			ExitCode: ExGuestError,
+			Advice:   "Your minikube certs likely expired, as a workaround, clear your minikube home dir `minikube delete --all --purge`",
+			Issues:   []int{10948},
+		},
+		Regexp: re(`wait 6m0s for node: wait for healthy API server: controlPlane never updated to`),
```
Comment (medyagh) [inline pkg/minikube/reason/known_issues.go:1025]: I am not convinced this is the right regex for this

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/reason/known_issues.go (hunk 3 above)
Comment (spowelljr) [reply pkg/minikube/reason/known_issues.go]: Removed

**Review by medyagh — APPROVED** [review]

**Pull request merged**